influent-transport/src/main/java/influent/internal/util/Inet4Network.java

```diff
@@ -0,0 +1,41 @@
+package influent.internal.util;
+
+import java.net.InetAddress;
+import java.util.BitSet;
+
+public class Inet4Network implements InetNetwork {
+  private BitSet netmask;
+  private BitSet network;
+
+  public Inet4Network(InetAddress base, int netmask) {
+    BitSet subnetmask = new BitSet(32);
+    subnetmask.set(32 - netmask, 32);
+    // base.getAddress() returns like [192, 168, 1, 10]
+    // but BitSet.valueOf() expects [10, 1, 168, 192]
```

Comment:
What does this means?

Reply:
BitSet represents little endianness, InetAddress.getBytes() returns network byte order (big endianness).

See
* https://docs.oracle.com/javase/jp/8/docs/api/java/net/InetAddress.html#getAddress--
* https://docs.oracle.com/javase/jp/8/docs/api/java/util/BitSet.html#valueOf-byte:A-

Reply:
When using BigSet#valueOf, the order of bytes follows the original one, but bits in each byte are reversed.

```scala
scala> BitSet.valueOf(Array[Byte](192.toByte, 168.toByte, 1, 10))
res15: java.util.BitSet = {6, 7, 11, 13, 15, 16, 25, 27}
```

This is
00000011-00010101-10000000-01010000

But `Array[Byte](192.toByte, 168.toByte, 1, 10)` is
11000000-00010101-10101000-00001010

So we should reverse the result of `InetAddress.getaddress` at bit level.
However, if such a operation is required, `BitSet` may not be useful……

Reply:
Hmm, what shoud we do?

1. Revert current implementation
1. Keep current implementation
1. Implement BitSet like class to handle `InetAddress.getAddress` w/o reversing it
1. Use some library support both IPv4 and IPv6

Reply:
I think 1 is better.

Both the previous implementation and the current one require bitwise operations.
It was my idea that BitSet makes subnet check easy, but it was wrong.

I will write property based tests and ensure that your implementation is correct, so there is no problem even if any corner cases exist.

Sorry for the  confusion.

Reply:
OK, I will revert BitSet related code.

> Sorry for the confusion.

No, problem!

Reply:
I've reverted BitSet related code.